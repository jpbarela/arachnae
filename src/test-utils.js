import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "react-jss";
import { defaultTheme } from "./Theme";

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
