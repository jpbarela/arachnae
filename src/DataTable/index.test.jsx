// @flow
import * as React from "react";
import { render, screen } from "../test-utils";
import { DataTable } from "./index";
import userEvent from "@testing-library/user-event";

describe("DataTable", () => {
  it("renders", () => {
    const container = render(
      <DataTable
        columns={[
          {
            header: "Strings",
            textAlign: "left",
          },
          {
            header: "Numbers",
            textAlign: "center",
            width: "8rem",
          },
          {
            header: "Dates",
            textAlign: "right",
            width: "25rem",
          },
        ]}
        data={[
          ["a", 1, new Date(2021, 0, 1).toDateString()],
          ["b", 2, new Date(2021, 1, 1).toDateString()],
          ["c", 3, new Date(2021, 2, 1).toDateString()],
        ]}
      />
    );
    expect(container.baseElement.firstChild).toMatchSnapshot();
  });

  it("sorts the data", () => {
    render(
      <DataTable
        columns={[
          {
            header: "Strings",
            textAlign: "left",
            sortable: true,
          },
          {
            header: "Numbers",
            textAlign: "center",
            width: "8rem",
          },
          {
            header: "Dates",
            textAlign: "right",
            width: "25rem",
          },
        ]}
        data={[
          ["a", 1, new Date(2021, 0, 1).toDateString()],
          ["b", 2, new Date(2021, 1, 1).toDateString()],
          ["c", 3, new Date(2021, 2, 1).toDateString()],
        ]}
      />
    );

    const sortIndicator = screen.getByLabelText("sortable");
    expect(sortIndicator).toBeInTheDocument();
    userEvent.click(sortIndicator.closest("div"));
    const ascendingSortIndicator = screen.getByLabelText("ascending sort");
    expect(ascendingSortIndicator).toBeInTheDocument();
    userEvent.click(ascendingSortIndicator);
    const descendingSortIndicator = screen.getByLabelText("descending sort");
    expect(descendingSortIndicator).toBeInTheDocument();
  });

  it("resets ascending on multi column sorts", () => {
    render(
      <DataTable
        columns={[
          {
            header: "Strings",
            textAlign: "left",
            sortable: true,
          },
          {
            header: "Numbers",
            textAlign: "center",
            width: "8rem",
          },
          {
            header: "Dates",
            textAlign: "right",
            width: "25rem",
            sortable: true,
          },
        ]}
        data={[
          ["a", 1, new Date(2021, 0, 1).toDateString()],
          ["b", 2, new Date(2021, 1, 1).toDateString()],
          ["c", 3, new Date(2021, 2, 1).toDateString()],
        ]}
      />
    );

    const sortIndicators = screen.getAllByLabelText("sortable");
    expect(sortIndicators.length).toBe(2);
    const firstButton = sortIndicators[0].closest("div");
    const secondButton = sortIndicators[1].closest("div");
    userEvent.click(firstButton);
    expect(screen.getByLabelText("ascending sort")).toBeInTheDocument();
    userEvent.click(secondButton);
    expect(screen.getByLabelText("ascending sort")).toBeInTheDocument();
  });
});
