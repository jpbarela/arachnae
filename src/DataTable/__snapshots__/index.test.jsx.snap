// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable renders 1`] = `
<div>
  <table
    class="table-0-2-2 table-d0-0-2-5"
  >
    <thead>
      <tr>
        <th
          class="tableHeader-0-2-4 tableHeader-d5-0-2-10"
          colspan="1"
          rowspan="1"
        >
          Strings
           
        </th>
        <th
          class="tableHeader-0-2-4 tableHeader-d8-0-2-13"
          colspan="1"
          rowspan="1"
        >
          Numbers
           
        </th>
        <th
          class="tableHeader-0-2-4 tableHeader-d11-0-2-16"
          colspan="1"
          rowspan="1"
        >
          Dates
           
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="tableData-0-2-3 tableData-d13-0-2-18"
          colspan="1"
          rowspan="1"
        >
          a
        </td>
        <td
          class="tableData-0-2-3 tableData-d16-0-2-21"
          colspan="1"
          rowspan="1"
        >
          1
        </td>
        <td
          class="tableData-0-2-3 tableData-d19-0-2-24"
          colspan="1"
          rowspan="1"
        >
          Fri Jan 01 2021
        </td>
      </tr>
      <tr>
        <td
          class="tableData-0-2-3 tableData-d22-0-2-27"
          colspan="1"
          rowspan="1"
        >
          b
        </td>
        <td
          class="tableData-0-2-3 tableData-d25-0-2-30"
          colspan="1"
          rowspan="1"
        >
          2
        </td>
        <td
          class="tableData-0-2-3 tableData-d28-0-2-33"
          colspan="1"
          rowspan="1"
        >
          Mon Feb 01 2021
        </td>
      </tr>
      <tr>
        <td
          class="tableData-0-2-3 tableData-d31-0-2-36"
          colspan="1"
          rowspan="1"
        >
          c
        </td>
        <td
          class="tableData-0-2-3 tableData-d34-0-2-39"
          colspan="1"
          rowspan="1"
        >
          3
        </td>
        <td
          class="tableData-0-2-3 tableData-d37-0-2-42"
          colspan="1"
          rowspan="1"
        >
          Mon Mar 01 2021
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
